src/db.cpp

```diff
@@ -2639,7 +2639,11 @@ bool redisDbPersistentData::insert(char *key, robj *o, bool fAssumeNew, dict_ite
         ensure(key);
     dictEntry *de;
     int res = dictAdd(m_pdict, key, o, &de);
-    serverAssert(FImplies(fAssumeNew, res == DICT_OK));
+    if (!FImplies(fAssumeNew, res == DICT_OK)) {
+        serverLog(LL_WARNING,
+            "Assumed new key %s existed in DB.", key);
+    }
+    // serverAssert(FImplies(fAssumeNew, res == DICT_OK));
```

Comment:
Please remove the old assert and we can merge.

Reply:
@JohnSully , I have removed the old assertion, thanks.

Reply:
@JohnSully Please review and merge the change by @karthyuom .  Thank you.